### ISSUE TITLE
RecipeViewer ore generation doesn't update when modifying ores through KJS

### ISSUE THREAD

**Issue by 3x1t-5tyl3** [GregTechCEu/GregTechCEu-1.19#1543]
### Checked for existing issues

- [X] I have checked for existing issues, and have found none.

### Tested latest version

- [X] I have checked that this occurs on the latest version.

### GregTech CEu Version

1.3.1 for 1.21

### Recipe Viewer Installed

EMI

### Environment

All

### Cross-Mod Interaction

Yes

### Other Installed Mods
<details>
<summary> ModList </summary>

- accelerated-decay-neoforge-21.0.0,
- AE2NetworkAnalyzer-1.21-1.0.1-neoforge,
- ae2wtlib-19.0.7-alpha,
- AI-Improvements-1.21-0.5.3,
- appleskin-neoforge-mc1.21-3.0.3,
- appliedenergistics2-19.0.8-alpha,
- architectury-13.0.3-neoforge,
- athena-neoforge-1.21-4.0.0,
- balm-neoforge-1.21-21.0.11-all,
- betterdays-1.21-3.1.0.5-NEOFORGE,
- bigger_ae2-1.21-1.4.0,
- block_variants-1.21-6.0,
- bucketlib-neoforge-1.21-4.0.0.4,
- cardboardboxes-1.21-0.1.2,
- cc-tweaked-1.21-forge-1.111.0,
- ceramicbucket-neoforge-1.21-6.0.0.0,
- cherishedworlds-neoforge-10.0.0+1.21,
- chisel-neoforge-1.21-1.8.1,
- clienttweaks-neoforge-1.21-21.0.2,
- cloth-config-15.0.127-neoforge,
- Clumps-neoforge-1.21-18.0.0.1,
- compact_storage-1.21-neoforge-8.0.74,
- configured-neoforge-1.21-2.5.0,
- Controlling-neoforge-1.21-18.0.1,
- cookingforblockheads-neoforge-1.21-21.0.3,
- cookyourfood--mc1.21--neoforge--1.12,
- cosmeticarmorreworked-1.21-v1-neoforge,
- craftingtweaks-neoforge-1.21-21.0.3,
- CreativeCore_NEOFORGE_v2.11.34_mc1.21,
- CTM-1.21-1.2.0,
- defaultoptions-neoforge-1.21-21.0.5,
- dynamiccrosshair-8.0.2+1.21-neoforge,
- elevatorid-1.21-1.10.1,
- embeddium-1.0.2+mc1.21,
- emi-1.1.8+1.21+neoforge,
- exitiantweaks-1.0.0,
- ExtremeSoundMuffler-3.47-neoforge-1.21,
- factory_blocks+neoforge+1.21-1.3.2,
- fallingleaves-1.21-2.4.0-beta.1,
- ferritecore-7.0.0-neoforge,
- flatbedrock-neoforge-87.0.0,
- flickerfix-1.21.0,
- foodtxf-1.21-1.6.3-neoforge,
- ftb-chunks-neoforge-2100.1.0,
- ftb-essentials-neoforge-2100.1.0,
- ftb-filter-system-neoforge-3.0.0,
- ftb-library-neoforge-2100.1.2,
- ftb-quests-neoforge-2100.1.2,
- ftb-teams-neoforge-2100.1.0,
- ftb-xmod-compat-neoforge-3.2.1,
- FuelInfo-2.1+1.21,
- getittogetherdrops-neoforge-1.21-1.3.1,
- GlitchCore-neoforge-1.21-2.0.0.2,
- globalpacks-neoforge-1.21-21.0.1,
- Glodium-1.21-1.2-neoforge,
- gtceu-1.21-1.3.1,
- hardcorerevival-neoforge-1.21-21.0.3,
- inventorysorter-1.21-24.0.4,
- invtweaks-1.21.0-1.1.1,
- ironladders-1.21-2.2.3-neoForge,
- Jade-1.21-NeoForge-15.1.3,
- jei-1.21-neoforge-19.0.0.14,
- journeymap-neoforge-1.21-6.0.0-beta.13,
- kleeslabs-neoforge-1.21-21.0.2,
- kubejs-2100.7.0-build.56,
- ldlib-neoforge-1.21-1.0.25.u,
- lmft-1.0.4+1.21-neoforge,
- logicchips-1.21-2.1.2-neoforge,
- MineMenu-neoforge-1.21-1.14.2,
- MobCatcher-NeoForge-1.21-1.5.2,
- modelfix-1.21-1.6,
- modernfix-neoforge-5.18.3+mc1.21,
- MouseTweaks-neoforge-mc1.21-2.26,
- NoChatReports-NEOFORGE-1.21-v2.8.0,
- okzoomer-neoforge-1.21.0-4.0.1,
- overloadedarmorbar-neoforge-1.21-1,
- packetfixer-neoforge-1.4.1-1.21-to-1.21,
- polylib-2100.0.3-build.153-neoforge,
- ProbeJS-7.1.1,
- reap-neoforge-1.21-1.1.1,
- rhino-2100.2.5-build.32,
- Searchables-neoforge-1.21-1.0.1,
- seasonhud-neoforge-1.21-1.8.9,
- SereneSeasons-neoforge-1.21-10.0.0.2,
- shrink-2.0.0.41-neoforge,
- spark-1.10.73-neoforge,
- Spice of Life Onion_NEOFORGE_v1.2.7_mc1.21,
- tia-1.21-1.2-neoforge,
- tiab-neoforge-1.21-6.0.0,
- toms_storage-1.21-2.0.3,
- ToolBelt-1.21-2.2.1,
- ToughAsNails-neoforge-1.21-10.0.0.5,
- trashslot-neoforge-1.21-21.0.2,
- trofers-neoforge-8.0.0,
- Unexperienced-1.21-2.0.0,
- utilitarian-1.21.0-0.10.1,
- VanillaIcecreamFix-1.2.2-beta,
- xtonesreworked-1.0.2-NF-1.21_21.0.0-beta,
- yeetusexperimentus-neoforge-87.0.0,

</details>

### Expected Behavior

The ore to show adjusted values instead of original values

### Actual Behavior

Shows original (unmodified) values

### Steps to Reproduce

Use this KJS script
```js
GTCEuServerEvents.oreVeins((event) => {
	event.modifyAll((veinid, vein) => {
		vein.dimensions("minecraft:overworld");
	});
});
```

### Additional Information

Not quite sure. Potentially because it doesn't reload them properly when modified *after* modifying?

Guessing it's either how it's registered with EMI or how it's registered with KJS.
The following script works however:
(I'm using modifyall here to loop through all of them so I don't have to make a list)
```js
GTCEuServerEvents.oreVeins((event) => {
	event.modifyAll((veinid, vein) => {
		event.modify(veinid, (vein2) => {
			vein2.dimensions("minecraft:overworld");
		});
	});
});
```

TL;DR ModifyAll doesn't work. Modify does work
